### PR TITLE
Updated extension endpoints OAS definitions

### LINKED ISSUE #20373: OAS incorrect extension paths
### Describe the Bug

The OAS definition still renders the old `/extension/displays` endpoints which no longer matches the current situation

https://docs.directus.io/reference/system/extensions.html
image

### To Reproduce

 Request the OpenAPI Specification and look at the `/extensions/*` paths.

### Directus Version

10.7.2

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by br41nslug**
Fixes #20373

## Scope

What's changed:

- Brought the OAS definition for the `/extensions/*` endpoints in sync again with the current state and docs

## Review Notes / Questions

- I would like to lorem ipsum

### CODE REVIEW

File: packages/specs/src/openapi.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-openapi: 3.0.1
+openapi: 3.1.0
```
Comment (paescuj) [inline packages/specs/src/openapi.yaml:1]: This should be updated in a separate PR
```suggestion
openapi: 3.0.1
```

**Review by paescuj — APPROVED** [review]

File: packages/specs/src/openapi.yaml (hunk 1 above)
Comment (br41nslug) [reply packages/specs/src/openapi.yaml]: Hmmm perhaps, i updated it to satisfy the openapi spec version regex 🤔 
image

Comment (paescuj) [reply packages/specs/src/openapi.yaml]: Yeah, we should update (and probably fix that) soon 😄

Comment (br41nslug) [reply packages/specs/src/openapi.yaml]: A follow up PR for that it is 👌
